internal/controller/backend_security_policy.go

```diff
@@ -89,6 +90,26 @@ func (c *BackendSecurityPolicyController) reconcile(ctx context.Context, bsp *ai
 		if err != nil {
 			return res, err
 		}
+		secretName, useGeneratedSecret := getBSPGeneratedSecretName(bsp)
+		if useGeneratedSecret {
+			var secret *corev1.Secret
+			secret, err = rotators.LookupSecret(ctx, c.client, bsp.Namespace, secretName)
+			if err != nil {
+				panic(fmt.Errorf("BUG: failed to lookup backend security policy secret %s/%s: %w", bsp.Namespace, bsp.Name, err))
```

Comment:
it looks like we hit this in tests/controller. you can revert this to an error (still no idea why it's happening though).